src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -284,15 +282,6 @@ void CoreWorkerDirectTaskReceiver::HandlePushTask(
           }
         }
       }
-      // If we spilled any return objects to plasma, notify the raylet to pin them.
-      // The raylet will then coordinate with the caller to manage the objects'
-      // lifetimes.
-      // TODO(edoakes): the plasma objects could be evicted between creating them
-      // here and when raylet pins them.
-      if (!plasma_return_ids.empty()) {
-        RAY_CHECK_OK(
-            local_raylet_client_->PinObjectIDs(caller_address, plasma_return_ids));
```

Comment:
Note that the change to do this in `core_worker.cc` means we're now making one RPC per spilled return ID. IMO this is not a concern but I can change it back if need be - it just makes this already complicated logic messier.